sycl/source/detail/kernel_compiler/kernel_compiler_sycl.cpp

```diff
@@ -9,59 +9,26 @@
 #include "kernel_compiler_sycl.hpp"
 #include <sycl/exception.hpp> // make_error_code
 
-#if __GNUC__ && __GNUC__ < 8
-
-// std::filesystem is not availalbe for GCC < 8
-// and much of the  cross-platform file handling code depends upon it.
-// Given that this extension is experimental and that the file
-// handling aspects are most likely temporary, it makes sense to
-// simply not support GCC<8.
```

Comment:
According to this comment DPC++ doesn't support GCC 7 and older.

> [SYCL] kernel_compiler remove GCC < 8 workarounds.

I would change the PR description to say that this patch adds support for GCC 7 rather than just removes the GCC< 8 workarounds. It's hard to say which GCC version is guaranteed to work, but I think GCC 7 is the first version supporting most of C++17 features. C++17 support is the main requirement for SYCL host compiler.